### PR TITLE
Handle single unsupported device

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

If there is a single unsupported device connected and an id isn't specified, filter it out of the list. handles the case where we display:

```
jonahwilliams-macbookpro2:foo jonahwilliams$ flutter run
Downloading Web SDK...                                              0.9s
Running "flutter pub get" in foo...                                 1.6s
No application found for TargetPlatform.web_javascript.
To add web support to a project, run `flutter create --web .`.
```

In an app that has web enabled but no web support when there are no other devices, supported or unsupported left.

### CODE REVIEW

File: packages/flutter_tools/lib/src/device.dart [hunk 1]
```diff
@@ -190,6 +190,12 @@ class DeviceManager {
           if (isDeviceSupportedForProject(device, flutterProject))
             device
       ];
+    } else if (devices.length == 1 &&
```
Comment (dnfield) [inline packages/flutter_tools/lib/src/device.dart]: What will this do if I'm in a flutter project that doesn't have desktop or web folders, but I have the options on in config?

**Review by dnfield — APPROVED** [review]
LGTM with question

File: packages/flutter_tools/lib/src/device.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/device.dart]: Both options?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/device.dart]: it should display "no supported devices for project"

Comment (zanderso) [inline packages/flutter_tools/lib/src/device.dart]: Could you just change the `devices.length > 1` above to `>=` ?

**Pull request merged**